deepfence_worker/utils/registry.go

```diff
@@ -108,11 +109,52 @@ func GetCredentialsFromRegistry(ctx context.Context, registryId string) (regCred
 		return jfrogCreds(reg, aes)
 	case constants.ECR:
 		return ecrCreds(reg, aes)
+	case constants.GITLAB:
+		return gitlabCreds(reg, aes)
 	default:
 		return regCreds{}, nil
 	}
 }
 
+func gitlabCreds(reg postgresql_db.GetContainerRegistryRow, aes encryption.AES) (regCreds, error) {
+	var (
+		err       error
+		hub       gitlab.RegistryGitlab
+		nonsecret gitlab.NonSecret
+		secret    gitlab.Secret
+	)
+
+	err = json.Unmarshal(reg.NonSecret, &nonsecret)
+	if err != nil {
+		log.Error().Msg(err.Error())
+	}
+	err = json.Unmarshal(reg.EncryptedSecret, &secret)
+	if err != nil {
+		log.Error().Msg(err.Error())
+	}
+
+	hub = gitlab.RegistryGitlab{
+		Name:      reg.Name,
+		Secret:    secret,
+		NonSecret: nonsecret,
+	}
+
+	err = hub.DecryptSecret(aes)
+	if err != nil {
+		log.Error().Msg(err.Error())
+	}
+
+	return regCreds{
+		URL:           hub.NonSecret.GitlabRegistryURL,
+		UserName:      "gitlab-ci-token",
+		Password:      hub.Secret.GitlabToken,
+		NameSpace:     "",
+		ImagePrefix:   "registry.gitlab.com",
+		SkipTLSVerify: false,
```

Comment:
this needs to be true in case of self signed certificate, required as long as we dont implement a way to upload self-signed certificates

Reply:
currently we are not supporting self signed certificates, but will change once we have that feature.